protocol/contracts/lpp/src/lpp.rs

```diff
@@ -891,6 +891,135 @@ mod test {
         assert_eq!(withdraw, Coin::new(1110));
     }
 
+    #[test]
+    fn open_and_repay_with_overflow() {
```

Comment:
This is not a unit test - it is an unnecessarily broad functional one that does not focus on the test case. Instead, it deals, unnecessarily, with persistence, configuration, access rights, and even with a whole liquidity pool. 
We need to test only Total!
Its length - that smells a lot!